pkg/reduce/readloop/readloop.go

```diff
@@ -150,6 +152,8 @@ func (rl *ReadLoop) Process(ctx context.Context, messages []*isb.ReadMessage) {
 					rl.log.Errorw("Failed to write message", zap.Any("msgOffSet", m.ReadOffset.String()), zap.String("partitionID", partitionID.String()), zap.Any("attempt", attempt), zap.Error(rErr))
 					return false, nil
 				}
+				// TODO - add real vertex name as metric dimension.
+				wal.EntriesCount.With(map[string]string{metricspkg.LabelVertex: "vertex-name"}).Inc()
```

Comment:
You can actually add this in the Write() method of WAL package, that would be cleaner and only expose to WAL.